### PR TITLE
Improve docker build stages

### PR DESCRIPTION
**Pull request by kondziorex7**
Use `turbo prune` functionality to exclude unnecessary packages and dependencies from final package.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -10,3 +10,7 @@ node_modules
 .vscode
 .idea
 .*.swp
+
+# Docker
+.dockerignore
```
Comment (imxeno) [inline .dockerignore:15]: Aren't dotfiles ignored by default, or at least not added if you don't explicitly state you want them?

File: Dockerfile.backend [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
-COPY . /src/
-RUN cd /src/ && \
-  yarn install && \
-  yarn build:backend
+FROM node:18 as pruner
+WORKDIR /src/
+COPY . .
+RUN yarn global add turbo
+RUN yarn prune:backend
+# RUN turbo prune --docker @l2beat/backend
```
Comment (imxeno) [inline Dockerfile.backend:8]: I guess we should either remove this comment or uncomment it and remove the prune:backend script. It's so specific to docker it won't ever be used anywhere else (for purposes other than debugging docker issues).

File: Dockerfile.backend [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+
+FROM node:18 as builder
+ARG ARCH
+WORKDIR /src/
+COPY --from=pruner /src/out/json/ /src/tsconfig.json ./
```
Comment (imxeno) [inline Dockerfile.backend:13]: Is tsconfig at this stage?

File: Dockerfile.backend [hunk 4]
```diff
@@ -1,10 +1,20 @@
-FROM node:18 as builder
 ARG ARCH=x86_64
 
-COPY . /src/
-RUN cd /src/ && \
-  yarn install && \
-  yarn build:backend
+FROM node:18 as pruner
```
Comment (imxeno) [inline Dockerfile.backend:3]: Is there a reason we don't use an alpine linux images?

**Review by imxeno — APPROVED** [review]

File: Dockerfile.backend (hunk 4 above)
Comment (kondziorex7) [reply Dockerfile.backend:3]: No, as discussed offline difftastic may be actually broken after such migration. I'll test it separately to this PR

File: Dockerfile.backend (hunk 3 above)
Comment (kondziorex7) [reply Dockerfile.backend]: Moved to post-install/pre-build copy

File: Dockerfile.backend (hunk 2 above)
Comment (kondziorex7) [reply Dockerfile.backend]: makes sense, also comment was left here unintentionally. Fixed

**Pull request merged**